[WIP] Rework again

eneiluj opened the pull request:
Hey! Maps app is alive, aliiiive.

Discussion started in #32 and #39 (there's a task list in this one).

This branch is based on `rework`. It's far from a first release but it's functional. Device management is coming soon from `eneiluj-devices` branch.

PhpUnit tests are working but are almost empty. CI script has been fixed. Makefile has been fixed...

jancborchardt commented:
Great great stuff @eneiluj! :) Here’s a first design review, and I’m focusing on the basic maps stuff first so we get that right:

- The maximum zoom-out level is too much, as you can see further than the edges of the world ;)
![maps app zoom out too much](https://user-images.githubusercontent.com/925062/56421900-0b497480-62a5-11e9-8dd5-d856686868e3.png)
   This is roughly what would be maximally needed, to simply fit the viewport:
![Maps app zoom max needed](https://user-images.githubusercontent.com/925062/56421899-0ab0de00-62a5-11e9-9f5d-6b572e3df155.png)
- The map ends in the pacific and it’s not possible to go from e.g. New Zealand to Venezuela. It should be possible to go around, so it’s not eurocentric:
![Maps app map ends in pacific and not possible to go over](https://user-images.githubusercontent.com/925062/56421898-0ab0de00-62a5-11e9-8a24-071b70c33f02.png)
- Zoom and geolocation buttons could be adjusted to Nextcloud button style (`class="button"`). Also they could move to the bottom right so to not take too much focus away from search:
![Maps app zoom and geolocation buttons could be adjusted to Nextcloud style](https://user-images.githubusercontent.com/925062/56421937-37fd8c00-62a5-11e9-8c90-54403deb62f2.png)
- Both layer switchers at the same time are not needed, only the one which is not used at the moment. Also they are both duplicated in the "More layers" box, which is not necessary:
![Maps app both layer switchers at same time not needed and duplicated in more layers menu](https://user-images.githubusercontent.com/925062/56421897-0ab0de00-62a5-11e9-9c1f-37bed35e296b.png)
- The lat/long info on the bottom left is not needed:
![Maps app geolocation info not necessary](https://user-images.githubusercontent.com/925062/56421895-0a184780-62a5-11e9-9be8-9770ab578ecd.png)
- The time bar on the bottom is only needed when a layer with time info is actually displayed. Otherwise it can be hidden:
![Maps app time bar only needed when layer with time info displayed](https://user-images.githubusercontent.com/925062/56421896-0ab0de00-62a5-11e9-827f-00bedeb07bc0.png)
- The size ruler is too present. It could be on the bottom right after the OSM credits – check Google Maps to compare:
![Maps app size ruler should be bottom left and fade out after some seconds fade in when zooming](https://user-images.githubusercontent.com/925062/56421894-0a184780-62a5-11e9-9b51-44053a166b18.png)
![Google Maps data bottom right](https://user-images.githubusercontent.com/925062/56422101-1fda3c80-62a6-11e9-8bad-ce5f1695736c.png)
- The search box could use an inline `icon-search` instead of the big button:
![Maps app global search box should be used](https://user-images.githubusercontent.com/925062/56421893-097fb100-62a5-11e9-8899-24f76fd3c159.png)
- Also, routing should just add a second field instead of using a separate box on the top right – compare with Google Maps for example:
![Maps app routing](https://user-images.githubusercontent.com/925062/56422201-8a8b7800-62a6-11e9-9dd7-2616980495e2.png)
   They also use the search box contents as _destination_ by default. (osm.org uses it as the starting point, which makes less sense):
![Google Maps search](https://user-images.githubusercontent.com/925062/56422274-d807e500-62a6-11e9-8568-f420b240bad0.png)
![Google Maps routing](https://user-images.githubusercontent.com/925062/56422273-d76f4e80-62a6-11e9-8733-783d87c7d886.png)
- The right click action menu is pretty core to the app but is not so discoverable. The issue here is that there need to be clickable elements on the map to interact with. Is that possible?
![Maps app right click not discoverable](https://user-images.githubusercontent.com/925062/56421891-097fb100-62a5-11e9-89e9-3fdfb5b34fe2.png)
- Then lastly the layers I didn’t look into so much yet. Mostly I was confused about when a layer is actually displayed. The eye indicator is not clear and relies on color change for indication which is not sufficient. I’d recommend to show the entries slightly greyed / with reduced transparency by default, and then every activated layer could show with its entry at full opacity and with the primary color bar on the left. Then the eye indicator is not needed:
![Maps app not sure when a layer is displayed eye indicator is unclear](https://user-images.githubusercontent.com/925062/56421892-097fb100-62a5-11e9-9ed7-2b68c06661b5.png)


What do you think @eneiluj? :) As said I first focused on the core map basics. All layer fanciness is of course awesome and integrates with the rest of Nextcloud, but we need to get the core stuff right first. :)

Also what do you think @nextcloud/maps @v1r0x @nextcloud/designers?

eneiluj commented:
Thanks for the feedback @jancborchardt ! I completely agree base map aspects should be fixed before talking about bigger features.

* map view does not have limits for movement and display anymore
* minimum zoom is now 2
* zoom control is now in the bottom right corner
* osm and aerial are not listed in layer selector anymore
* aerial layer button is visible when any other layer is selected
* osm button is visible when aerial layer is selected
* mouse position control (coordinates in bottom left corner) has been removed (even if I loved it :wink:)
* scale control has been moved to bottom left. it's quite complex to integrate it with the provider credits... maybe later
* search submit button is nicer

Now about the rest:

* zoom and geolocation buttons: i gave up after 5 minutes (booo) trying to make them look more nextcloudish. May the PR gods be with us.
* time filter slider is actually already supposed to appear in `var(--color-primary)` but i guess you didn't enable any information layer so it stayed in its initial state where there sometimes is a color bug. this bug is now fixed
* hiding time filter slider: i put this in my todo list, i still need to harmonize filtering between all js controllers...
* search: i didn't touch anything, it's still in `rework` state. there's a lot to do there. for the moment it only searches for addresses. it does not interact with information layers. todo again.
* routing: i just use a leaflet plugin with very small modifications. with a bit of effort it can elegantly be combined with the search form.....todo :grin: let's discuss that in a specific issue
* context menu: i'm not sure i get what you wrote. do you mean it's not very useful unless there are elements to "rightclick" on? for the moment, it only triggers actions related to where the right-click was made. i'll check if the leaflet plugin allows to make a difference between rightclick on the map and on a marker/line.
* toggling layers: i made it like that to postpone the reflection. i agree we could get rid of the eye. problem is: some menu entries are foldable. what should happen when we fold one? should it disable the layer? if so, what if the user wants to fold it but keep the layer visible?
* non localized photos: this is to be changed. the warning color means the coordinates are not trustable yet because they were guessed. do you think this should become a top-level menu entry?

and this is just the visible part of the mapsberg. a lot of work in perspective! :tada: :computer:

e-alfred commented:
This PR works really good and is already pretty awesome, but I found a few problems with the app:

- It seems to load content from http servers which lead to mixed content warnings/errors if the server is run with https (probably almost all Nextcloud servers out there hopefully):

`Loading mixed (insecure) display content “http://a.tile.stamen.com/watercolor/11/2345/4526.jpg” on a secure page`

- The track parser has some problems with a few *.gpx files created by GPSLogger and the spinner loads forever. In the browser console I see these messages:

```
XML Parsing Error: not well-formed
Location: 
Line Number 1, Column 171:

XML Parsing Error: unclosed token
Location: https://mylink.com/apps/maps/#
Line Number 2176, Column 1: mylink.com:2176:1
Error: Invalid XML: <?xml version="1.0" encoding="UTF-8" ?><gpx version="1.0" creator="GPSLogger 101 - http://gpslogger.mendhak.com/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.topografix.com/GPX/1/0" xsi:schemaLocation="http://www.topografix.com/GPX/1/0 http://www.topografix.com/GPX/1/0/gpx.xsd"><time>2019-04-17T09:42:58.593Z</time><trk><trkseg><trkpt lat="23.3500434" lon="43.1943276"><ele>612.7000122070312</ele><time>2019-04-17T09:42:58.593Z</time><src>network</src></trkpt>
<trkpt lat="23.3500127" lon="43.1943589"><ele>612.7000122070312</ele><time>2019-04-17T09:43:03.552Z</time><src>network</src></trkpt>
<trkpt lat="23.35000504" lon="43.19433953"><ele>621.0</ele><time>2019-04-17T09:43:06.774Z</time><course>309.2</course><speed>1.48</speed><src>gps</src></trkpt>
<trkpt lat="23.3499923" lon="43.19433325"><ele>626.0</ele><time>2019-04-17T09:43:07.774Z</time><course>280.2</course><speed>0.85</speed><geoidheight>45.0</geoidheight><src>gps</src><sat>14</sat><hd… core.js:2:1821
```

- If I try to enable the pictures view I get an error message "Failed to load photos" and "Failed to load non-geolocalized photos" respectively. If I execute `php occ maps:scan-photos` in verbose mode no output is shown whatsoever. In my log I found this error if I activate the photos view:

```
{"reqId":"XLrihkjfQKzYSLMAaRnTgQAAAAo","level":3,"time":"2019-04-20T11:12:40+02:00","remoteAddr":"1.1.1.1","user":"user","app":"index","method":"GET","url":"\/nc\/apps\/maps\/photos\/nonlocalized","message":{"Exception":"Doctrine\\DBAL\\Exception\\InvalidFieldNameException","Message":"An exception occurred while executing 'SELECT * FROM `oc_maps_photos` where `user_id` = ? and (`lat` is null or `long` is  null)' with params [\"user\"]:\n\nSQLSTATE[42S22]: Column not found: 1054 Unknown column 'long' in 'where clause'","Code":0,"Trace":[{"file":"\/var\/www\/nc\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/DBALException.php","line":128,"function":"convertException","class":"Doctrine\\DBAL\\Driver\\AbstractMySQLDriver","type":"->","args":["An exception occurred while executing 'SELECT * FROM `oc_maps_photos` where `user_id` = ? and (`lat` is null or `long` is  null)' with params [\"user\"]:\n\nSQLSTATE[42S22]: Column not found: 1054 Unknown column 'long' in 'where clause'",{"errorInfo":["42S22",1054,"Unknown column 'long' in 'where clause'"],"__class__":"Doctrine\\DBAL\\Driver\\PDOException"}]},{"file":"\/var\/www\/nc\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Statement.php","line":177,"function":"driverExceptionDuringQuery","class":"Doctrine\\DBAL\\DBALException","type":"::","args":[{"__class__":"Doctrine\\DBAL\\Driver\\PDOMySql\\Driver"},{"errorInfo":["42S22",1054,"Unknown column 'long' in 'where clause'"],"__class__":"Doctrine\\DBAL\\Driver\\PDOException"},"SELECT * FROM `oc_maps_photos` where `user_id` = ? and (`lat` is null or `long` is  null)",{"1":"user"}]},{"file":"\/var\/www\/nc\/lib\/public\/AppFramework\/Db\/Mapper.php","line":256,"function":"execute","class":"Doctrine\\DBAL\\Statement","type":"->","args":[]},{"file":"\/var\/www\/nc\/lib\/public\/AppFramework\/Db\/Mapper.php","line":344,"function":"execute","class":"OCP\\AppFramework\\Db\\Mapper","type":"->","args":["SELECT * FROM `*PREFIX*maps_photos` where `user_id` = ? and (`lat` is null or `long` is  null)",["user"],null,null]},{"file":"\/var\/www\/nc\/apps\/maps\/lib\/DB\/GeophotoMapper.php","line":49,"function":"findEntities","class":"OCP\\AppFramework\\Db\\Mapper","type":"->","args":["SELECT * FROM `*PREFIX*maps_photos` where `user_id` = ? and (`lat` is null or `long` is  null)",["user"],null,null]},{"file":"\/var\/www\/nc\/apps\/maps\/lib\/Service\/GeophotoService.php","line":95,"function":"findAllNonLocalized","class":"OCA\\Maps\\DB\\GeophotoMapper","type":"->","args":["user"]},{"file":"\/var\/www\/nc\/apps\/maps\/lib\/Controller\/PhotosController.php","line":52,"function":"getNonLocalizedFromDB","class":"OCA\\Maps\\Service\\GeophotoService","type":"->","args":["user"]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":166,"function":"getNonLocalizedPhotosFromDb","class":"OCA\\Maps\\Controller\\PhotosController","type":"->","args":[]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":99,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Maps\\Controller\\PhotosController"},"getNonLocalizedPhotosFromDb"]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/App.php","line":118,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Maps\\Controller\\PhotosController"},"getNonLocalizedPhotosFromDb"]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Maps\\Controller\\PhotosController","getNonLocalizedPhotosFromDb",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"maps.photos.getNonLocalizedPhotosFromDb"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"maps.photos.getNonLocalizedPhotosFromDb"}]},{"file":"\/var\/www\/nc\/lib\/private\/Route\/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"maps.photos.getNonLocalizedPhotosFromDb"}]},{"file":"\/var\/www\/nc\/lib\/base.php","line":987,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/apps\/maps\/photos\/nonlocalized"]},{"file":"\/var\/www\/nc\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/var\/www\/nc\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Driver\/AbstractMySQLDriver.php","Line":71,"Previous":{"Exception":"Doctrine\\DBAL\\Driver\\PDOException","Message":"SQLSTATE[42S22]: Column not found: 1054 Unknown column 'long' in 'where clause'","Code":"42S22","Trace":[{"file":"\/var\/www\/nc\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Statement.php","line":168,"function":"execute","class":"Doctrine\\DBAL\\Driver\\PDOStatement","type":"->","args":[null]},{"file":"\/var\/www\/nc\/lib\/public\/AppFramework\/Db\/Mapper.php","line":256,"function":"execute","class":"Doctrine\\DBAL\\Statement","type":"->","args":[]},{"file":"\/var\/www\/nc\/lib\/public\/AppFramework\/Db\/Mapper.php","line":344,"function":"execute","class":"OCP\\AppFramework\\Db\\Mapper","type":"->","args":["SELECT * FROM `*PREFIX*maps_photos` where `user_id` = ? and (`lat` is null or `long` is  null)",["user"],null,null]},{"file":"\/var\/www\/nc\/apps\/maps\/lib\/DB\/GeophotoMapper.php","line":49,"function":"findEntities","class":"OCP\\AppFramework\\Db\\Mapper","type":"->","args":["SELECT * FROM `*PREFIX*maps_photos` where `user_id` = ? and (`lat` is null or `long` is  null)",["user"],null,null]},{"file":"\/var\/www\/nc\/apps\/maps\/lib\/Service\/GeophotoService.php","line":95,"function":"findAllNonLocalized","class":"OCA\\Maps\\DB\\GeophotoMapper","type":"->","args":["user"]},{"file":"\/var\/www\/nc\/apps\/maps\/lib\/Controller\/PhotosController.php","line":52,"function":"getNonLocalizedFromDB","class":"OCA\\Maps\\Service\\GeophotoService","type":"->","args":["user"]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":166,"function":"getNonLocalizedPhotosFromDb","class":"OCA\\Maps\\Controller\\PhotosController","type":"->","args":[]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":99,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Maps\\Controller\\PhotosController"},"getNonLocalizedPhotosFromDb"]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/App.php","line":118,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Maps\\Controller\\PhotosController"},"getNonLocalizedPhotosFromDb"]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Maps\\Controller\\PhotosController","getNonLocalizedPhotosFromDb",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"maps.photos.getNonLocalizedPhotosFromDb"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"maps.photos.getNonLocalizedPhotosFromDb"}]},{"file":"\/var\/www\/nc\/lib\/private\/Route\/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"maps.photos.getNonLocalizedPhotosFromDb"}]},{"file":"\/var\/www\/nc\/lib\/base.php","line":987,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/apps\/maps\/photos\/nonlocalized"]},{"file":"\/var\/www\/nc\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/var\/www\/nc\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Driver\/PDOStatement.php","Line":107,"Previous":{"Exception":"PDOException","Message":"SQLSTATE[42S22]: Column not found: 1054 Unknown column 'long' in 'where clause'","Code":"42S22","Trace":[{"file":"\/var\/www\/nc\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Driver\/PDOStatement.php","line":105,"function":"execute","class":"PDOStatement","type":"->","args":[null]},{"file":"\/var\/www\/nc\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Statement.php","line":168,"function":"execute","class":"Doctrine\\DBAL\\Driver\\PDOStatement","type":"->","args":[null]},{"file":"\/var\/www\/nc\/lib\/public\/AppFramework\/Db\/Mapper.php","line":256,"function":"execute","class":"Doctrine\\DBAL\\Statement","type":"->","args":[]},{"file":"\/var\/www\/nc\/lib\/public\/AppFramework\/Db\/Mapper.php","line":344,"function":"execute","class":"OCP\\AppFramework\\Db\\Mapper","type":"->","args":["SELECT * FROM `*PREFIX*maps_photos` where `user_id` = ? and (`lat` is null or `long` is  null)",["user"],null,null]},{"file":"\/var\/www\/nc\/apps\/maps\/lib\/DB\/GeophotoMapper.php","line":49,"function":"findEntities","class":"OCP\\AppFramework\\Db\\Mapper","type":"->","args":["SELECT * FROM `*PREFIX*maps_photos` where `user_id` = ? and (`lat` is null or `long` is  null)",["user"],null,null]},{"file":"\/var\/www\/nc\/apps\/maps\/lib\/Service\/GeophotoService.php","line":95,"function":"findAllNonLocalized","class":"OCA\\Maps\\DB\\GeophotoMapper","type":"->","args":["user"]},{"file":"\/var\/www\/nc\/apps\/maps\/lib\/Controller\/PhotosController.php","line":52,"function":"getNonLocalizedFromDB","class":"OCA\\Maps\\Service\\GeophotoService","type":"->","args":["user"]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":166,"function":"getNonLocalizedPhotosFromDb","class":"OCA\\Maps\\Controller\\PhotosController","type":"->","args":[]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":99,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Maps\\Controller\\PhotosController"},"getNonLocalizedPhotosFromDb"]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/App.php","line":118,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Maps\\Controller\\PhotosController"},"getNonLocalizedPhotosFromDb"]},{"file":"\/var\/www\/nc\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Maps\\Controller\\PhotosController","getNonLocalizedPhotosFromDb",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"maps.photos.getNonLocalizedPhotosFromDb"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"maps.photos.getNonLocalizedPhotosFromDb"}]},{"file":"\/var\/www\/nc\/lib\/private\/Route\/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"maps.photos.getNonLocalizedPhotosFromDb"}]},{"file":"\/var\/www\/nc\/lib\/base.php","line":987,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/apps\/maps\/photos\/nonlocalized"]},{"file":"\/var\/www\/nc\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/var\/www\/nc\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Driver\/PDOStatement.php","Line":105}},"CustomMessage":"--"},"userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:66.0) Gecko\/20100101 Firefox\/66.0","version":"15.0.7.0"}
```

jancborchardt commented:
> Thanks for the feedback @jancborchardt ! I completely agree base map aspects should be fixed before talking about bigger features.
> 
> * map view does not have limits for movement and display anymore
> * minimum zoom is now 2
> * zoom control is now in the bottom right corner
> * osm and aerial are not listed in layer selector anymore
> * aerial layer button is visible when any other layer is selected
> * osm button is visible when aerial layer is selected
> * mouse position control (coordinates in bottom left corner) has been removed (even if I loved it wink)
> * scale control has been moved to bottom left. it's quite complex to integrate it with the provider credits... maybe later
> * search submit button is nicer

Veeeery nice! :)


> Now about the rest:
> 
> * zoom and geolocation buttons: i gave up after 5 minutes (booo) trying to make them look more nextcloudish. May the PR gods be with us.

Right, I will try to look into this at some point. It’s not that big of a deal for now. :)


> * time filter slider is actually already supposed to appear in `var(--color-primary)` but i guess you didn't enable any information layer so it stayed in its initial state where there sometimes is a color bug. this bug is now fixed

Yep, seems to be fixed now. Although it seems to be a lightened color-primary? And when I pull the slider on either side almost to the end, the slider color goes orange again.


> * hiding time filter slider: i put this in my todo list, i still need to harmonize filtering between all js controllers...

:+1: 


> * search: i didn't touch anything, it's still in `rework` state. there's a lot to do there. for the moment it only searches for addresses. it does not interact with information layers. todo again.
> * routing: i just use a leaflet plugin with very small modifications. with a bit of effort it can elegantly be combined with the search form.....todo grin let's discuss that in a specific issue

Yes, makes sense, it’s a bigger thing. :)

> * context menu: i'm not sure i get what you wrote. do you mean it's not very useful unless there are elements to "rightclick" on? for the moment, it only triggers actions related to where the right-click was made. i'll check if the leaflet plugin allows to make a difference between rightclick on the map and on a marker/line.

I mean that the menu is _very_ useful, but that people don’t necessarily know how to get to it as right-click is not such a common thing on Maps. I for example never ever used right click on Google Maps until just yesterday to test if they have it. And all the functionality in there is also available on simple clicking of the map.

> * toggling layers: i made it like that to postpone the reflection. i agree we could get rid of the eye. problem is: some menu entries are foldable. what should happen when we fold one? should it disable the layer? if so, what if the user wants to fold it but keep the layer visible?

Folding/unfolding is only toggled on clicking the triangle on the left, not the whole entry. So clicking the entry should toggle it (make it `opacity: 1;` and add the `.active` bar on the left). And folding is separate from being active.

> * non localized photos: this is to be changed. the warning color means the coordinates are not trustable yet because they were guessed. do you think this should become a top-level menu entry?

I think it should either just be part of the normal photos layer, or not shown at all. It’s fine if coordinates are guessed I think, and just show them as if normal. If something is wrong they can be edited.


> and this is just the visible part of the mapsberg. a lot of work in perspective! :tada: :computer:


:tada: !

---------------------

And here come some additional things. :)

- @v1r0x wanted to port the Maps app to Vue.js at some point. Maybe before we build more and more stuff in this is a good point to do it? Or what do you think @nextcloud/javascript?
- It’s still possible to scroll out of the map on the top and bottom. This is not necessary, and the grey part outside of the world doesn’t ever need to be visible. It can just end there
![Maps app still possible to scroll out of the map](https://user-images.githubusercontent.com/925062/56455656-32fe1280-6361-11e9-8589-4b3681fe56fe.png)
- The geolocation button could move to the bottom right too as the other controls now are. :)
![Geolocation button should be bottom right too](https://user-images.githubusercontent.com/925062/56455655-32fe1280-6361-11e9-9c4c-88fabf9f27f7.png)
- Clicking the geolocation button directly jumps to your location and a very close zoom level (scale shows 10m). Ideally there should be some sort of quick animation and then not zoom in so far, more like the scale showing 100m.
- The satellite view has no labels or anything at all. It’s kind of confusing as you have no sense of space with that. Is it possible to half-overlay the street map, or show important structural stuff like cities and roads so you have some reference?
![Satellite view has no labels or anything at all](https://user-images.githubusercontent.com/925062/56455654-32fe1280-6361-11e9-8ca6-2f5995096d0f.png)
- Searching for a place moves the map only so it’s barely visible in the corner. It should center it in the viewport instead:
![Searching for a place should center it in the viewport](https://user-images.githubusercontent.com/925062/56455653-32657c00-6361-11e9-998e-122d84232217.png)
- In the detail popover of a favorite there’s too many controls: The fields are not clear as they only have icons and no text. I’d say only "Name" is needed for now for simplicy (also with placeholder "Name" instead of "no name". And then the "Cancel" button duplicates the x in the top right. Also, move is not necessary as you could simply drag the icon on the map directly. And the remaining "Save" and "Delete" buttons: Save should be on the right, and have `.icon-checkmark` instead of a floppy disk. :)
![Cancel button duplicates x and fields are not clear - only name is enough for now - and move could just be dragging the fav symbol](https://user-images.githubusercontent.com/925062/56455652-32657c00-6361-11e9-8777-086d71adf3b5.png)
- Bonus: in the additional layers I would love to have an inverted map like that since it’s not so centric on the northern hemisphere ;)
![inverted map](https://user-images.githubusercontent.com/925062/56455730-7e64f080-6362-11e9-9b6b-76a3719473c4.png)
- Also – a bigger topic, but – our map is of course very skewed due to its projection, showing central Europe and northern US bigger than they actually are related to places close to the equator like African countries. Google Maps solves this by at some point switching to a globe view, which I don’t know if we can do? :D But just for reference also [the true size of Africa](http://kai.sub.blue/en/africa.html)
![Maps globe](https://user-images.githubusercontent.com/925062/56455787-32ff1200-6363-11e9-8ded-0f6c9b00eccd.png)
![](http://kai.sub.blue/images/true_size_of_africa_v3-da3dd38e.jpg)

jancborchardt commented:
Oh and btw, on a build there are these untracked files:
```
composer.lock
package-lock.json
vendor/
```

They should either be committed or in .gitignore, not sure which is best practice. :)

e-alfred commented:
I found another problem regarding the non-localized images showing up in the log causing an error 500:

{"reqId":"XLr12W03pdK5Bjmdksl6kwAAABg","level":3,"time":"2019-04-20T12:35:06+02:00","remoteAddr":"1.1.1.1","user":"user","app":"PHP","method":"GET","url":"\/nc\/apps\/maps\/photos\/nonlocalized","message":"Trying to get property 'trk' of non-object at \/var\/www\/nc\/apps\/maps\/lib\/Service\/GeophotoService.php#179","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:66.0) Gecko\/20100101 Firefox\/66.0","version":"15.0.7.0"}
{"reqId":"XLr12W03pdK5Bjmdksl6kwAAABg","level":3,"time":"2019-04-20T12:35:06+02:00","remoteAddr":"1.1.1.1","user":"user","app":"PHP","method":"GET","url":"\/nc\/apps\/maps\/photos\/nonlocalized","message":"Invalid argument supplied for foreach() at \/var\/www\/nc\/apps\/maps\/lib\/Service\/GeophotoService.php#179","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:66.0) Gecko\/20100101 Firefox\/66.0","version":"15.0.7.0"}
`{"reqId":"XLr12W03pdK5Bjmdksl6kwAAABg","level":3,"time":"2019-04-20T12:35:05+02:00","remoteAddr":"1.1.1.1","user":"user","app":"PHP","method":"GET","url":"\/nc\/apps\/maps\/photos\/nonlocalized","message":"simplexml_load_string(): &lt;trkpt lat=&quot;22.32452 at \/var\/www\/nc\/apps\/maps\/lib\/Service\/GeophotoService.php#178","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:66.0) Gecko\/20100101 Firefox\/66.0","version":"15.0.7.0"}`

eneiluj commented:
@e-alfred and @jancborchardt : wow thanks for the precise and fast feedback!

@e-alfred:

* watercolor tile server address is now in https
* contacts: All%20contacts must be translated to correctly get to the contact view... Is there another way (locale independent) to get a link to a contact?
* could you provide one or two problematic gpx files?
* photo query issue is now fixed. my fault, i didn't fully review the code from @tacruc.

@jancborchardt:

* about the slider:
    * Color was primary-light. It's now primary
    * The warning color means the slider range is going to be dynamically changed when the user drops the slider outside its current max range. It's doing the same when the interval becomes too small.
    * The slider behaviour is a big topic :wink:. I don't know how to improve it but it needs improvement.
* context menu:
    * I forgot to tell you about the style: I didn't make any modification to leaflet-contextmenu plugin. CSS can easily be overridden but html structure is another story
    * As there is no action that is only available in the context menu for the moment, i'm ok if the user does not "find" the context menu... If at some point we add interesting features in this context menu, we'll have to make it obvious it exists...
* folding/toggling: Ok fine by me. It's been done and pushed. A small addition: When a top level menu entry disabled and folded, clicking on it enables it AND unfold it. What do you think about that?
* non-localized photos: This is also a big topic we can discuss in a specific issue. I agree it needs efforts :smile:
* untracked files: they shouldn't be tracked because they are generated/modified for local needs. I've updated `.gitignore` file

I'll have a look to the rest of your comments later :wink:. Thanks again. We're in a nice dynamic! :heart: :nerd_face:

skjnldsv commented:
> * In the detail popover of a favorite there’s too many controls: The fields are not clear as they only have icons and no text. I’d say only "Name" is needed for now for simplicy (also with placeholder "Name" instead of "no name". And then the "Cancel" button duplicates the x in the top right. Also, move is not necessary as you could simply drag the icon on the map directly. And the remaining "Save" and "Delete" buttons: Save should be on the right, and have `.icon-checkmark` instead of a floppy disk. :)

ALso, please follow our design guidelines regarding a popovermenu like that! :)

eneiluj commented:
@skjnldsv Yep, will do. Thanks for the remarks.

skjnldsv commented:
> * Bonus: in the additional layers I would love to have an inverted map like that since it’s not so centric on the northern hemisphere ;)

I would say it's really not a priority though. I'm not really sure it's even a useful feature. Do people really use that? We should not confuse the user with potential unused options as well. Let's see what OSM and GoogleMap offers and see what we can do to keep the unified UX the user is used to? :)

skjnldsv commented:
> * Bonus: in the additional layers I would love to have an inverted map like that since it’s not so centric on the northern hemisphere ;)

I would say it's really not a priority though. I'm not really sure it's even a useful feature. Would people really use that? We should not confuse the user with potential unused options as well. Let's see what OSM and GoogleMap offers and see what we can do to keep the unified UX the user is used to? :)

@eneiluj impressive work btw! The file change and additions/removal is really impressive! :tada:

jancborchardt commented:
> > * Bonus: in the additional layers I would love to have an inverted map like that since it’s not so centric on the northern hemisphere ;)
> 
> I would say it's really not a priority though. I'm not really sure it's even a useful feature. Would people really use that? We should not confuse the user with potential unused options as well. Let's see what OSM and GoogleMap offers and see what we can do to keep the unified UX the user is used to? :)

I know it’s not a priority – that suggestion is just because maps are traditionally racist. ;)

---------

> context menu:
> I forgot to tell you about the style: I didn't make any modification to leaflet-contextmenu plugin. CSS can easily be overridden but html structure is another story

Right, maybe just a bit of additional padding for every entry is enough for better spacing then. :)

> folding/toggling: Ok fine by me. It's been done and pushed. A small addition: When a top level menu entry disabled and folded, clicking on it enables it AND unfold it. What do you think about that?

:+1: 

You rock! :) Agree, it’s going along nicely! Maybe before it gets too big, soon we should just merge to master, even if it’s still work in progress and then work on topic branches?

skjnldsv commented:
> that suggestion is just because maps are traditionally racist and it would be a nice touch. ;)

Any good reading on the subject? First time I'm hearing this.
(Let me set my comment as off topic to not flood) :)

e-alfred commented:
@eneiluj One thing that is missing from the context menu is the ability to remove a contact from the map. Currently only adding one is possible.

@skjnldsv If you are interested, you can start here: https://en.wikipedia.org/wiki/Gall%E2%80%93Peters_projection It is a complicated and wide-ranging topic, though.

eneiluj commented:
@jancborchardt:

* i'll have a look at vue.js soon
* map max bounds are now [-720, 720] for longitude and [-90, 90] for latitude as it's not possible to only restrict latitude AFAIK
* locate control:
    * has been moved
    * now flies to current location when enabled
    * now follows current location until user manually pans (moves the map)
    * remembers map's state before being enabled and restore it when disabled
    * does not zoom further than 15
    * there still is a problem: accuracy circle is drawn before flyTo animation so it's becoming bigger during the animation... I'll check if it's possible to extend locate control to fix that.
* aerial view labels: It's possible to add an overlay layer with such information. We just have to find one with the information we want. I'll check that.
* search: there is now a flyTo animation to search result
* favorite popup: I'll improve that soon. I don't know about allowing such elements to be dragged... As user most likely drags the map very often, there could be a lot of involuntary favorite moves. What do you think?
* inverted map: I'll leave this feature to the gods of PRs :wink:
* map globe: It is possible with webGL, I've seen such maps in web pages. Problem is: it's rendered by libraries that are not fully featured like Leaflet. We can't use a globe as our main map AFAIK. I think it's possible to change the world projection in Leaflet though. https://github.com/kartena/Proj4Leaflet . I'm not sure how easy it is to use it with our tile providers.

> Maybe before it gets too big, soon we should just merge to master

Yeah sure (as it's already too big :wink:). I can keep working on rework-again anyway and create per-feature branches like you said. rework-again would be kind of my master.

@e-alfred:

* reset contact geolocation information: why not? Let me think about that. Maybe it belongs more in the "contact" context menu which is not implemented yet (I don't even know if it's possible with leaflet-contextmenu)
* I need a problematic GPX file to debug the buggy parsing.

e-alfred commented:
> reset contact geolocation information: why not? Let me think about that. Maybe it belongs more in the "contact" context menu which is not implemented yet (I don't even know if it's possible with leaflet-contextmenu)

Not all contacts have location information available and it is possible to add a contact to a certain place using the context menu, but there is no option to remove the contact anywhere.

A few things I noticed:

- Is it possible to allow different modes of transport for the routing engine?
- Sometimes the results by entering a city or town name leads to no routing result and an error is returned, but the user gets no response from the UI about that. Here is an error from the browser console:

```
Routing error: 
Object { message: "HTTP request failed: undefined", url: "https://router.project-osrm.org/route/v1/driving/143.34960937500003,52.37559917665913;143.026504609508,50.15947595?overview=false&alternatives=true&steps=true&hints=;", status: -1, target: XMLHttpRequest }
leaflet-routing-machine.min.js:14:26552
```

Aside from that, the search for places is a bit difficult to use because there are multiple results at times but not all work to get a routing between these places (see the error message above).

jancborchardt commented:
Very good stuff, awesome progress @eneiluj! :) Added 1 measly commit of my own, and have another round of feedback. ;D

- [ ] During the build I get these warnings, should this be fixed centrally somewhere or is it with my setup? 
```
/home/jan/.nvm/versions/node/v11.3.0/bin/npm install
npm WARN leaflet.markercluster@1.4.1 requires a peer of leaflet@~1.3.1 but none is installed. You must install peer dependencies yourself.
npm WARN opening_hours@3.5.0 requires a peer of country-language@^0.1.7 but none is installed. You must install peer dependencies yourself.
npm WARN opening_hours@3.5.0 requires a peer of i18next-client@^1.11.1 but none is installed. You must install peer dependencies yourself.
npm WARN opening_hours@3.5.0 requires a peer of moment@^2.10.6 but none is installed. You must install peer dependencies yourself.
```
- [ ] Also during the build there seems to be a high severity vulnerability:
```
found 1 high severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
sed -i 's/L\.Browser\.touch/L.Browser.mobile/g' node_modules/leaflet.elevation/dist/Leaflet.Elevation-0.0.2.min.js
```

Onto more design stuff:

## 🗺 Map view
- [ ] The popover for search results could follow the style that the favorites popover also does. :) (Using the proper spacing, Nextcloud font, adjust width etc). Also an "Add to favorites" button would be great there! :woman_singer: 
![search result popover style](https://user-images.githubusercontent.com/925062/56469934-459b4900-6440-11e9-9a2b-49ee81268baa.png)
- [ ] The "Roads and labels" are only useful for the satellite and watercolor layers (otherwise it duplicates labels), and should just be shown always for those 2 layers as they are quite useless without:
![Roads and labels are only useful for watercolor and satellite should just be shown there no option for the others](https://user-images.githubusercontent.com/925062/56469649-a7f24a80-643c-11e9-8e9c-dfe7c49e7589.png)
- [ ] The different styles of icons is a bit strange: Could we make the marker also flat – simply in blue – just like the app icon? And the markers themselves need a slight shadow as they are otherwise difficult to see, and a white star in the middle instead of transparency would be better for visibility too. Could the grouped black marker be colored according to the fav color they are grouping?
![Maps app relation of icon sizes a bit strange](https://user-images.githubusercontent.com/925062/56469650-a7f24a80-643c-11e9-99c7-fc4dba0c06bd.png)
![markers need slight shadow otherwise difficult to see and grouped dark marker should be colored accordingly too](https://user-images.githubusercontent.com/925062/56469640-a6c11d80-643c-11e9-9091-fa5494b5e071.png)
- [ ] The scale ruler on the bottom left currently shows both km and miles. It could show either or depending on the user’s locale instead:
![scale control could show km or miles not both depending on locale](https://user-images.githubusercontent.com/925062/56469638-a6288700-643c-11e9-9ac0-e960c6ae7b38.png)

## ⭐ Favorites
- [ ] The favorites layer also shows the time slider on the bottom, but it’s not really necessary for that. Favorites is a default layer which will most likely be active all the time, but the time slider is not something that should always be shown. (Am also thinking if the time slider being on the map is really the best for the other layers too – might be more obvious if it’s in the left navigation instead, below the respective layer. But we can look at that separately. :)
- [ ] In Favorites, the first "no category" category could instead be called "Personal", just like we name the first calendar etc. All new favorites could by default go into that category. Also, if only that category exists, it doesn’t need to be shown as a subentry of favorites:
![Favorites category first category should be called personal, and does not need to be shown if no other categories exist](https://user-images.githubusercontent.com/925062/56469646-a759b400-643c-11e9-88bb-e894ad50a0fb.png)
- [ ] The colored hover tooltips are difficult to see and read. They would be better as regular white Nextcloud tooltips / popovers instead, showing their content to the top just like when you then click on the favorite:
![hover tooltips difficult to see should be regular white Nextcloud tooltips instead](https://user-images.githubusercontent.com/925062/56469665-f273c700-643c-11e9-8d6e-76c6b6e2e868.png)
- [ ] Importing a gpx file could automatically add a new category with the name of the file and should also show the category directly. Currently it’s not obvious what happened:
![Importing file should automatically add new category with the name of the file and also show the category directly](https://user-images.githubusercontent.com/925062/56469643-a6c11d80-643c-11e9-9c39-d9238b3351c0.png)
- [ ] Imported favorites have their name duplicated as comment:
![imported favorites have name as comment duplicated](https://user-images.githubusercontent.com/925062/56469666-f30c5d80-643c-11e9-9b7f-1427bdbc8672.png)
- [ ] I’d say let’s remove the "Export displayed favorites" from the "Favorites" row, instead having an "Export" function for all of the individual categories.
- [ ] kmz files exported from Maps.me are not supported. Would be cool if we can do that, as I now needed to manually convert them: https://jwaghetti.blogspot.com/2014/02/converting-kmz-to-gpx.html
![kmz files like exported from Mapsme are not supported](https://user-images.githubusercontent.com/925062/56469644-a759b400-643c-11e9-8c22-fdf770c9ec35.png)
- [ ] Would be cool if clicking on gpx/kml/kmz files in the Files app could handle them in the Maps app instead of prompting to download: E.g. automatically adding them as a favorites category named after the file, or import the track if it is one:
![Handle kmz kml gpx bookmark files automatically add as favorites category and enable](https://user-images.githubusercontent.com/925062/56469645-a759b400-643c-11e9-98e3-8cf461808c27.png)

## 👥 Contacts
- Actually haven’t gotten that to run yet. Am not getting any errors or feedback. What am I doing wrong? :D Does it load really long, or do _all_ contacts need a correct address?

## ⭕ General
- [ ] This might rather be something for our standardization: The indentation of subentries with the additional bar floating in the air does not look good, and indentation is not really needed for the first level (see Mail app) cc @nextcloud/designers :)
![indentation does not look good better whitespace maybe - has to be changed in core instead](https://user-images.githubusercontent.com/925062/56469639-a6288700-643c-11e9-9b80-f3bb8792df02.png)
![Maps app menu indent bars too much](https://user-images.githubusercontent.com/925062/56469647-a759b400-643c-11e9-91e8-7c07eb2ce336.png)

eneiluj commented:
@jancborchardt I didn't touch the "search" part. It's still like in `rework` branch. There are maaaany things to do about it. I just didn't start :smile:.

* npm warnings: markercluster requires a precise version of leaflet but it's fine with latest (1.4.0). I don't know why opening_hours complains about dependencies. It's not consistent with information in https://www.npmjs.com/package/opening_hours?activeTab=dependencies . 
* I don't get any `high severity vulnerability`...

### :world_map: Map view

* the `sed` snippet fixes the only incompatibility between L.Elevation and recent leaflet versions... This plugin is very old but i didn't find a good replacement.
* search result popup: in my todo list
* Roads and labels is now automatically displayed when Aerial map or Watercolor is selected. It's not shown in layer selector anymore
* revolution around layer selection: Selector now has `icon-menu`. No more `mouseover`. When clicked, it hides osm/aerial buttons and shows layer list. If a layer is selected or the user clicks anywhere else, display goes back to "normal". I like it this way! Do you?
* favorite icon star is now white instead of transparent (I had a hard time on this one)
* I still can't put a box-shadow on single favorite icon... In my todo list.
* favorite cluster: As the cluster may contain favorites from multiple categories, it's not obvious what we should do. Now I've made it like photos and contacts: We take the first cluster element to represent the others. It could be ultra neat :grin: : Cluster color could be a mix of contained favorites colors! What do you think?
* scale units are now adapted to locale (imperial for GB, Ireland, New Zeland, USA, Australia, Canada)

### :star: Favorites

* Favorites have creation and modification dates. They can be filtered as any other element. Even if it's not a killer feature I think we should keep this. Imagine you register every new bike parking spot you see in your city, it could be nice to see the evolution through time.
* default category name is now "Personal" and is translatable
* As there is no interaction in favorite tooltip, don't you think it can stay smaller than a popover menu?
* I like the fact tooltips come with the same color than the favorite category. Could readability it be improved and still keep the category color?
* import/export: Category is saved in exported gpx file. I think it's good the way it is. It sounds clear to me, every favorite from the gpx file is strictly imported (with its category information). Ow if there is no category, for a favorite in a gpx file, file name could be used as a fallback value. What do you think?
* name duplicated in comment: I can't reproduce this bug.
* export category: I agree it's nice in terms of interface but what if I enable some categories, then I adjust time filter and want to export what I see? 
* import KMZ files: I'll make it possible. => todo. Could you give me an example file (I don't have any google account).
* files action: Hmmm such default action is provided by GpxPod because it has a single purpose but with Maps, there are multiple things you could do with a gpx/kml file like import favorites, load a track, import a device history etc...

### :busts_in_silhouette: Contacts 

Contacts part is very basic for the moment. You expected too much from it :wink:. It displays contacts which already have a "geo" information (https://www.evenx.com/vcard-3-0-format-specification). The map context menu allows you to place a contact (edit its vcard geo field). Then it's displayed.

Using the address is not that simple. It need one or two brainstorm sessions :smile:. When should the coordinates be deduced from the address?
* Client side, each time the contacts layer is loaded? => No because it's heavy and repetitive
* Server side? => Mmmmmaybe but when? If it's done for all contacts of all users, it's quite heavy.

To be continued

### ⭕ General

I agree it does not look good. Changing server css might be a better solution than dirty patching it in the apps :grin:.

@e-alfred:

* Routing is very basic for the moment. Don't expect too much :wink:. Yes it's possible to set routing options in [leaflet routing machine](http://www.liedman.net/leaflet-routing-machine/tutorials/alternative-routers/) like vehicle but it's only available with some routing engines like Graphhopper
* The failure you get is happening from times to times with the default "demo" OSRM server. The message you mentionned (the one printed in console logs) is saying this can happen. That's why this is not an option. The failure does not depend on the place you select but is rather random.

jancborchardt commented:
@eneiluj aaawesome! Have to go through your full reply still – now just for info: I used markdown checkboxes in my comment above so you can check off things for satisfaction. ;D

jancborchardt commented:
(Also, I edited the comment meanwhile, if you just checked it out via email notification. Bunch of additional review points e.g. for Photos showing. :)

eneiluj commented:
@jancborchardt Yeah I was going to check the checkboxes and I was caught up with something. That will be done :grin:.

jancborchardt commented:
@eneiluj I can only say "WOW!" about the progress. :) It's sooo cool!

I'd say it's your call, but when do you feel like it's good to merge this to master? I have some more feedback, but it's getting more small and nitpicky so stuff is really good! :)

eneiluj commented:
@jancborchardt Thanks for your enthusiasm! And also for your nice suggestion storm! :cyclone: :penguin: 

Yep it can be merged into master. Apparently I'm not allowed to do it. Would you prefer authorizing me or doing the merge yourself?

I added a first generated translation template in case transifex needs it.
Merge conflicts are just about l10n files and the README.

How would you like to work after the merge? Who would be the integrator? Should `rework-again` branch be deleted?

tacruc commented:
> ## :busts_in_silhouette: Contacts
> 
>     * Actually haven’t gotten that to run yet. Am not getting any errors or feedback. What am I doing wrong? :D Does it load really long, or do _all_ contacts need a correct address?


> ## :busts_in_silhouette: Contacts
> Contacts part is very basic for the moment. You expected too much from it wink. It displays contacts which already have a "geo" information (https://www.evenx.com/vcard-3-0-format-specification). The map context menu allows you to place a contact (edit its vcard geo field). Then it's displayed.
> 
> Using the address is not that simple. It need one or two brainstorm sessions smile. When should the coordinates be deduced from the address?
> 
>     Client side, each time the contacts layer is loaded? => No because it's heavy and repetitive
>     Server side? => Mmmmmaybe but when? If it's done for all contacts of all users, it's quite heavy.
> 
> To be continued

Lets move the contacts discussion into #34.

jancborchardt commented:
> Yep it can be merged into master. Apparently I'm not allowed to do it. Would you prefer authorizing me or doing the merge yourself?
> Merge conflicts are just about l10n files and the README.

I have to check how to do it myself, because of the conflicts. I would just force push it to master – or how t odo it? You should be able to do it too as I just removed master branch protection. Let me know if it doesn’t work and if you are ok with that, then we can do that.

> I added a first generated translation template in case transifex needs it.

:+1: 

> How would you like to work after the merge? Who would be the integrator? Should rework-again branch be deleted?

Yes, so how we usually work in Nextcloud apps is that master is generally always working, and we use small topic branches for individual fixes or features. All the rework-stuff can then be closed. I would also activate branch protection for the master branch so that 2 reviews are necessary for any pull request to be merged.

It seems the Maps team for now is you @eneiluj, @tacruc and me? So I would update that at @nextcloud/maps and make you both coordinators for the team too.

Does that all work? :)

eneiluj commented:
Ok let's do that! I'll force-push `rework-again` into `master` in 15 minutes (for obvious administrative and legal reasons :wink: or if someone prefers to properly merge with `master`).

Fine by me for keeping master functional and playing in this team with those rules.

eneiluj commented:
@jancborchardt `master` is still protected so I can't force push into it. I don't have access to project settings. Could you temporarily remove protection on master?

jancborchardt commented:
@eneiluj hmm, thought it was already removed, unset all the settings. Now removed it all again, could you try? :)

eneiluj commented:
@jancborchardt It worked. Thanks. You can put the protection back.

jancborchardt commented:
Gooood goood stuff! :) Put the branch protection back with 2 reviews required, we can also do 1 if 2 is too much.

I’m a bit busy with other stuff today and tomorrow probably too, so feel free to check more of the notes above – some day this week I will copy the remaining stuff over to issues, maybe first to ones grouped on topic (Routing, Photo display, Favorites etc) to keep it simple and have a unified vision. Does that work for you?

eneiluj commented:
Ow yeah!

I think one review is enough. If I create a PR, maybe only one of us will be available to review. Can someone approve his/her own PR? (it would be silly but i wonder...)

I'll slow down a bit on Maps. I left a bunch of stuff behind :grin:.

jancborchardt commented:
@eneiluj cool stuff! Edited it to 1 required review. And no, people can’t approve their own PR, as that would indeed be silly. :D 

Yeah, it’s good to relax a bit. :) Then when coming back to it, one sees more issues and where stuff could be simplified.

Btw @eneiluj do you already want to request a certificate for the app store (with ID "maps")? We should probably still do some polishing before publishing – and maybe we can also do it to coincide with a Nextcloud major release, but just to have that already done. :)

eneiluj commented:
@jancborchardt Yes I'll make a certificate request.

I'll be happy to publish the first release when we feel it's time, for a major NC release or not.

jancborchardt commented:
@eneiluj what we could do is publish it in conjunction with when the GNOME Maps favorites sync project is done and working. Just an idea – but when great work has been done, it also needs to be promoted with the appropriate greatness. ;)

eneiluj commented:
@jancborchardt  Let's hope a good favorite client is ready soon(ish). I would :heart: looooove to be able to sync favorites with OsmAnd. Being able to interact with such great projects as Gnome Maps and OsmAnd would be more than awesome! :smiley_cat: Make some noise! :wink:

e-alfred commented:
This app is already so awesome and has more features than the old Maps app so it should definitely be released to the App Store as soon as possible. Integration with other apps would be a huge bonus, maybe one day an Android companion app could be done as well?

v1r0x commented:
> It seems the Maps team for now is you @eneiluj, @tacruc and me? So I would update that at @nextcloud/maps and make you both coordinators for the team too.

Hey, I'm still here! :stuck_out_tongue_closed_eyes: 
I'm pretty busy, so can't help much on code side right now. But can help testing and commenting until I have more time for coding again :)

jancborchardt commented:
@e-alfred it’s better to wait a bit and polish it more, especially since there are so many features and things to check still. :) When people try a thing for the first time and are confused, they are not going to try it again so quickly.

> Hey, I'm still here! :stuck_out_tongue_closed_eyes:
> I'm pretty busy, so can't help much on code side right now. But can help testing and commenting until I have more time for coding again :)

Yaaaaay! :) Check out current master! All your work and @eneiluj’s on top is in there. :)
